src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -320,7 +320,22 @@ bool AsyncRPCOperation_sendmany::main_impl() {
         zOutputsDeque.push_back(o);
     }
 
-    
+    // When spending notes, take a snapshot of note witnesses and anchors as the treestate will
+    // change upon arrival of new blocks which contain joinsplit transactions.  This is likely
+    // to happen as creating a chained joinsplit transaction can take longer than the block interval.
+    if (z_inputs_.size() > 0) {
+        LOCK2(cs_main, pwalletMain->cs_wallet);
+        for (auto t : z_inputs_) {
+            JSOutPoint jso = std::get<0>(t);
+            std::vector<JSOutPoint> vOutPoints = { jso };
```

Comment:
I'm confused by this. `GetNoteWitnesses()` will only ever return a single witness per `JSOutPoint`. Why not fetch all witnesses in a single `GetNoteWitnesses()` call?